R/utils.R

```diff
@@ -12,70 +8,112 @@ phantom_run <- function(args, wait = TRUE) {
-#' found via the \code{PATH} variable.
+#'
+#' This function was designed primarily to help Windows users since it is
+#' cumbersome to modify the \code{PATH} variable. Mac OS X users may install
+#' PhantomJS via Homebrew. If you download the package from the PhantomJS
+#' website instead, please make sure the executable can be found via the
+#' \code{PATH} variable.
+#'
+#' On Windows, the directory specified by the environment variable
+#' \code{APPDATA} is used to store \file{phantomjs.exe}. On OS X, the directory
+#' \file{~/Library/Application Support} is used. On other platforms (such as
+#' Linux), the directory \file{~/bin} is used. If these directories are not
+#' writable, the directory \file{PhantomJS} under the installation directory of
+#' the \pkg{webshot} package will be tried. If this directory still fails, you
+#' will have to install PhantomJS by yourself.
 #' @param version The version number of PhantomJS.
 #' @return \code{NULL} (the executable is written to a system directory).
 #' @export
 install_phantomjs <- function(version = '2.1.1') {
-  if (!is_windows()) {
-    warning('This function is currently for Windows only')
-    return()
-  }
-
-  appdata <- Sys.getenv('APPDATA', NA)
-  if (is.na(appdata)) stop('The environment variable APPDATA is not set')
-  destdir <- file.path(appdata, 'PhantomJS')
-  dir.create(destdir, showWarnings = FALSE)
-
   owd <- setwd(tempdir())
   on.exit(setwd(owd), add = TRUE)
-  zipfile <- sprintf('phantomjs-%s-windows.zip', version)
-  link <- paste0('https://bitbucket.org/ariya/phantomjs/downloads/', zipfile)
-  download.file(link, zipfile, mode = 'wb')
-  utils::unzip(zipfile)
-  zipdir <- sub('.zip$', '', zipfile)
-  file.copy(file.path(zipdir, 'bin', 'phantomjs.exe'), destdir, overwrite = TRUE)
-  message('phantomjs.exe has been installed to ', normalizePath(destdir))
+  base <- 'https://bitbucket.org/ariya/phantomjs/downloads/'
+  if (is_windows()) {
+    zipfile <- sprintf('phantomjs-%s-windows.zip', version)
+    download.file(paste0(base, zipfile), zipfile, mode = 'wb')
+    utils::unzip(zipfile)
+    zipdir <- sub('.zip$', '', zipfile)
+    exec <- file.path(zipdir, 'bin', 'phantomjs.exe')
+  } else if (is_osx()) {
+    zipfile <- sprintf('phantomjs-%s-macosx.zip', version)
+    download.file(paste0(base, zipfile), zipfile, mode = 'wb')
+    utils::unzip(zipfile)
+    zipdir <- sub('.zip$', '', zipfile)
+    exec <- file.path(zipdir, 'bin', 'phantomjs')
+    Sys.chmod(exec, '0755')  # chmod +x
+  } else {
+    zipfile <- sprintf(
+      'phantomjs-%s-linux-%s.tar.bz2', version,
+      if (grepl('64', Sys.info()[['machine']])) 'x86_64' else 'i686'
+    )
+    download.file(paste0(base, zipfile), zipfile, mode = 'wb')
+    utils::untar(zipfile)
+    zipdir <- sub('.tar.bz2$', '', zipfile)
+    exec <- file.path(zipdir, 'bin', 'phantomjs')
+    Sys.chmod(exec, '0755')  # chmod +x
+  }
+  success <- FALSE
+  dirs <- phantom_paths()
```

Comment:
looks like I tested a couple minutes too early;  all clear now and working as expected.  Thanks.